### DISCUSSION TITLE
Combine API-token with username/recipient

### DISCUSSION THREAD

**Discussion by M0nk3yOo** [anonaddy/anonaddy#336] [Feature Requests]
Combine API with username/recipient.
For example, if I create a username (for ex. in Bitwarden) with token 1, it creates an alias for user1 (including recipient address).
if I create another username with token 2 (another instance or sth), it creates an alias for user2 (including recipient address).

Best
André

**Comment by willbrowningme** [comment]
Do you mean on the same AnonAddy account? 

So token A would create aliases and automatically attach recipient A?

And token B would create aliases and automatically attach recipient B?

**Reply by M0nk3yOo** [reply]
Yes, that's what I mean/suggest

**Reply by willbrowningme** [reply]
Not sure this is something I'll be adding as it would be quite complicated.

You can already set a default recipient for each different username on an account. If you're using the browser extension or a mobile app it only takes a couple of clicks to change the recipient upon creating an alias.

**Reply by M0nk3yOo** [reply]
Just an idea:

I don't know your DB structure, I could imagine that you've a separate API-table. What do you think about create another column in it and save a setting for which username (of course as DB-relation to the usernames-DB) the particular API-key is. (as like the setting username/default recipient). Or is this too simple in my mind because I don't know the DB structure?

**Reply by teanup** [reply]
Hi, any update on this? I would love to see this feature available
I need to assign recipients to aliases (on the same custom domain) based on the Bitwarden client (API key) that was used to generate them. Unfortunately I don't think that is possible as of now.
